packages/edit-site/src/components/global-styles/screen-revisions/index.js

```diff
@@ -55,13 +55,37 @@ function ScreenRevisions() {
 	const { setEditorCanvasContainerView } = unlock(
 		useDispatch( editSiteStore )
 	);
+	const selectedRevisionMatchesEditorStyles = areGlobalStyleConfigsEqual(
+		globalStylesRevision,
+		userConfig
+	);
+	const shouldSelectFirstItem =
+		! selectedRevisionMatchesEditorStyles &&
+		( ! selectedRevisionId || 'unsaved' === selectedRevisionId );
```

Comment:
`areGlobalStyleConfigsEqual` checks that `globalStylesRevision` matches `userConfig`, however the `useEffect` updates only the `globalStylesRevision`. Could there ever be a case when the `useEffect` goes into an infinite loop if the updated `globalStylesRevision` doesn't match the `userConfig`?

Or to put it differently, in the `useEffect` do we need to also check that the current `globalStylesRevision` is not equal to the first item in the list of revisions, to avoid a potential infinite loop? Apologies if I'm overthinking this!

Reply:
> Or to put it differently, in the useEffect do we need to also check that the current globalStylesRevision is not equal to the first item in the list of revisions, to avoid a potential infinite loop? 

Not overthinking at all. `! selectedRevisionId || 'unsaved' === selectedRevisionId` implies that the first item is selected by default, but it wouldn't hurt to elaborate this defence. I'll have a play.

Thanks again for thinking about this one!

Reply:
> areGlobalStyleConfigsEqual checks that globalStylesRevision matches userConfig, however the useEffect updates only the globalStylesRevision. Could there ever be a case when the useEffect goes into an infinite loop if the updated globalStylesRevision doesn't match the userConfig?

Thinking about that made me realize that the `selectedRevisionId` is 100% redundant since it's stored in (and saved and the same time as) `globalStylesRevision`.

So, here's what I have:

```js
	const shouldSelectFirstItem =
                // If there is a revision to load in the first place,
                !! firstRevision?.id &&

                // AND if a revision is loaded in the preview pane that is
                // different from the current editor state (userConfig),
		! selectedRevisionMatchesEditorStyles &&

                // AND if there's no currently-selected revision (a user hasn't clicked on anything)
                // we can go ahead and select the first revision.
		 ! currentlySelectedRevision?.id;
```

What I'm trying to express here is that, if nothing has been selected, but for some reason the current state of the revisions pane doesn't reflect what's in the editor, let's fix that by selecting the first item in the list. 

So, if globalStylesRevision doesn't match the userConfig and there's no revision selected in the list the `useEffect` condition will pass and then select the first revision using the state setter `setCurrentlySelectedRevision`.

The next time it spins around it will fail because it's selected a revision and we've double checked that an id exists for `firstRevision?.id`. 

The only thing I can think of is if `firstRevision?.id` is falsey. But if that happens we've got bigger problems 😄

Reply:
Sounds good to me, thanks for laying that out!